storage/conformance_test.go

```diff
@@ -16,21 +16,337 @@ package storage
+	}
+
+	rt.id = testRes.TestID
+
+	// Create wrapped client which will send emulator instructions
+	rt.host.Path = ""
+	client, err := wrappedClient(rt.T, rt.host.String(), rt.id)
+	if err != nil {
+		return fmt.Errorf("creating wrapped client: %v", err)
+	}
+	rt.wrappedClient = client
+
+	return nil
+}
+
+// Verify that all instructions for a given retry testID have been used up.
+func (rt *retrySubtest) check() error {
+	rt.host.Path = strings.Join([]string{"retry_test", rt.id}, "/")
+	c := http.DefaultClient
+	resp, err := c.Get(rt.host.String())
+	if err != nil || resp.StatusCode != 200 {
+		return fmt.Errorf("getting retry test: err: %v, resp: %+v", err, resp)
+	}
+	defer func() {
+		closeErr := resp.Body.Close()
+		if err == nil {
+			err = closeErr
+		}
+	}()
+	testRes := struct {
+		Instructions map[string][]string
+		Completed    bool
+	}{}
+	if err := json.NewDecoder(resp.Body).Decode(&testRes); err != nil {
+		return fmt.Errorf("decoding response: %v", err)
+	}
+	if !testRes.Completed {
+		return fmt.Errorf("test not completed; unused instructions: %+v", testRes.Instructions)
+	}
+	return nil
+}
+
+// Delete a retry test resource.
+func (rt *retrySubtest) delete() error {
+	rt.host.Path = strings.Join([]string{"retry_test", rt.id}, "/")
+	c := http.DefaultClient
+	req, err := http.NewRequest("DELETE", rt.host.String(), nil)
+	if err != nil {
+		return fmt.Errorf("creating request: %v", err)
+	}
+	resp, err := c.Do(req)
+	if err != nil || resp.StatusCode != 200 {
+		return fmt.Errorf("deleting test: err: %v, resp: %+v", err, resp)
+	}
+	return nil
+}
+
+type testRoundTripper struct {
+	*testing.T
+	rt     http.RoundTripper
+	testID string
+}
+
+func (wt *testRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	r.Header.Set("x-retry-test-id", wt.testID)
+
+	requestDump, err := httputil.DumpRequest(r, false)
+	if err != nil {
+		wt.Logf("error creating request dump: %v", err)
+	}
+
+	resp, err := wt.rt.RoundTrip(r)
+	if err != nil {
+		wt.Logf("roundtrip error (may be expected): %v\nrequest: %s", err, requestDump)
+	}
+	return resp, err
+}
+
+// Create custom client that sends instruction
+func wrappedClient(t *testing.T, host, testID string) (*Client, error) {
+	ctx := context.Background()
+	base := http.DefaultTransport
+	trans, err := htransport.NewTransport(ctx, base, option.WithScopes(raw.DevstorageFullControlScope),
+		option.WithUserAgent("custom-user-agent"))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create http client: %v", err)
+	}
+	c := http.Client{Transport: trans}
+
+	// Add RoundTripper to the created HTTP client
+	wrappedTrans := &testRoundTripper{rt: c.Transport, testID: testID, T: t}
+	c.Transport = wrappedTrans
+
+	// Supply this client to storage.NewClient
+	client, err := NewClient(ctx, option.WithHTTPClient(&c), option.WithEndpoint(host+"/storage/v1/"))
+	return client, err
+}
+
+// A url is only parsed correctly by the url package if it has a scheme,
```

Comment:
We shouldn't have to do this in our testing code-- if we expect a certain form for STORAGE_EMULATOR_HOST here we should be prescriptive about it.